### PR TITLE
refactor: upgraded frontend-build version to v12

### PR DESCRIPTION
**Pull request by BilalQamar95**
**TL;DR -** [Eslint was updated in `frontend-build` version resulting in it's version being bumped to v12. In order to reciprocate eslint version update, this PR updated `frontend-build` version to v12.0.0.]

GIT Ticket: [42: Upgrade eslint to v8.x](https://github.com/openedx/frontend-wg/issues/42)

**What changed?**

- Updated `frontend-build` to v12
- Resolved `eslint` issues

**Developer Checklist**
- [ ] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by muselesscreator** [comment]
We would much rather exclude some of these new and not-agreed-on eslint rules than update the style for our whole repo.
If there are new-mismatching rules, please set up ignores for them and create a ticket for us to review the new eslint rules.

**Review by muselesscreator — CHANGES_REQUESTED** [review]
We would much rather exclude some of these new and not-agreed-on eslint rules than update the style for our whole repo.
If there are new-mismatching rules, please set up ignores for them and create a ticket for us to review the new eslint rules.

**Comment by arbrandes** [comment]
@BilalQamar95, looks like @muselesscreator requested some changes.  Can we update the eslint rules for this repo to make the PR more palatable?

**Comment by BilalQamar95** [comment]
> @BilalQamar95, looks like @muselesscreator requested some changes. Can we update the eslint rules for this repo to make the PR more palatable?

@arbrandes, the comment from @muselesscreator triggered a thread on edx internal slack in result of which [there was a PR merged](https://github.com/openedx/frontend-build/pull/261) disabling some rules that were deemed unnecessary. This PR was updated afterwards, in light of those changes. If there are still new or not-agreed-on eslint rules in the PR I would be happy to revise those.

**Comment by BilalQamar95** [comment]
@arbrandes I have resolved all the merge conflicts.

**Review by arbrandes — APPROVED** [review]
+1
